retrofit2/src/main/java/com/linecorp/armeria/client/retrofit2/PipeBuffer.java

```diff
@@ -36,6 +36,11 @@ void write(byte[] source, int offset, int byteCount) {
         if (byteCount == 0) {
             return;
         }
+        /*
+         * Once use okhttp4, the synchronized block could be replaced by Reentrant and Condition for Virtual Thread support.
```

Comment:
Is there any plan to use okhttp4?
If that is not too far in the future, I hope to work in this PR.

Reply:
I think it's probably best to follow the okhttp version upstream is using which seems to be okhttp3 unless there is a significant benefit.
https://github.com/square/retrofit/blob/trunk/gradle/libs.versions.toml#L17

On a side note, does the okio side PR contain changes for APIs which Armeria is using? I'm asking because okhttp itself is an http engine and usage within Armeria is minimal.